src/readbin/fsi_benchmark.ml

```diff
@@ -0,0 +1,120 @@
+  let doc = "The testing stripped binary." in
+  Arg.(required & pos 1 (some non_dir_file) None & info [] ~docv:"binary" ~doc)
+
+let truth : string Term.t =
+  let doc =
+    "The ground truth. The ground truth can be an unstripped
+  binary, or a .scm file with symbol information. In .scm file, each symbol should
+  be in format of:
+  (<symbol name> <symbol start address> <symbol end address>), e.g.
+  (malloc@@GLIBC_2.4 0x11034 0x11038)" in
+  Arg.(required & pos 2 (some non_dir_file) None
+       & info [] ~docv:"ground-truth" ~doc)
+
+let print_metrics : metrics list Term.t =
+  let opts = [
+    "precision", `with_prec;
+    "recall", `with_recl;
+    "F_measure", `with_F;
+    "TP", `with_TP;
+    "FP", `with_FP;
+    "FN", `with_FN;
+  ] in
+  let doc = "Print metrics. User can choose to print -cprecision, -crecall,
+  -cF_measure, -cTP, -cFP and -cFN." in
+  Arg.(value & opt_all (enum opts) (List.map ~f:snd opts) &
+       info ["with-metrics"; "c"] ~doc)
+
+let output_metric_value oc {false_positive;false_negative;
+                            true_positive;prec;recl;f_05} metrics : unit =
+  let output_ratio = fprintf oc "\t%.2g" in
+  let output_int = fprintf oc "\t%d" in
+  List.iter metrics ~f:(function
+      | `with_prec -> output_ratio prec
+      | `with_recl -> output_ratio recl
+      | `with_F -> output_ratio f_05
+      | `with_TP -> output_int true_positive
+      | `with_FN -> output_int false_negative
+      | `with_FP -> output_int false_positive);
+  fprintf oc "\n"
+
+let string_of_metric = function
+  | `with_prec -> "Prcs"
+  | `with_recl -> "Rcll"
+  | `with_F -> "F_05"
+  | `with_TP -> "TP"
+  | `with_FN -> "FN"
+  | `with_FP -> "FP"
+
+let print oc tool result print_metrics : unit =
+  let tool_name = if tool = "bap-byteweight" then "BW" else "IDA" in
+  fprintf oc "Tool";
+  List.iter print_metrics ~f:(fun m -> Printf.fprintf oc "\t%s"
+                               @@ string_of_metric m);
+  fprintf oc "\n%s" tool_name;
+  output_metric_value oc result print_metrics
+
+let compare_against tool_name bin truth_name print_metrics : unit =
+  let open Or_error in
+  match (Func_start.of_tool tool_name ~testbin:bin >>| fun tool ->
+         Func_start.of_truth truth_name ~testbin:bin >>| fun truth ->
+         let result =
+           let false_positive = Set.(length (diff tool truth)) in
+           let false_negative = Set.(length (diff truth tool)) in
+           let true_positive = Set.length truth - false_negative in
+           let ratio x = Float.(of_int true_positive / (of_int true_positive + of_int x)) in
+           let prec = ratio false_positive in
+           let recl = ratio false_negative in
+           let f_05 = 1.5 *. prec *. recl /. (0.5 *. prec +. recl) in
+           {false_positive;false_negative;true_positive;prec;recl;f_05} in
+         print stdout tool_name result print_metrics) with
+  | Ok _ -> ()
+  | Error err ->
+    Format.printf "Function start is not recognized properly due to
```

Comment:
Why are you using at some places `Format.printf` and on other places `Printf.printf`? I usually just open either `Format` or `Printf`, and use whatever was brought into the scope.
... aha, I can answer on my question myself. That's because you want to use `Error.pp`. What about using `Format` module for all printfs?